lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java

```diff
@@ -207,7 +206,65 @@ private void launch(final YarnSubmissionFromCS yarnSubmission) throws IOExceptio
       } catch (InjectionException ie) {
         throw new RuntimeException("Unable to submit job due to " + ie);
       }
+      writeDriverHttpEndPoint(yarnSubmission.getDriverFolder(),
+          submissionHelper.getStringApplicationId(), jobFolderOnDFS.getPath());
+    }
+  }
+
+  /**
+   * We leave a file behind in job submission directory so that clr client can figure out
+   * the applicationId and yarn rest endpoint.
+   * @param driverFolder
+   * @param applicationId
+   * @throws IOException
+   */
+  private void writeDriverHttpEndPoint(final File driverFolder,
+                                       final String applicationId,
+                                       final Path dfsPath) throws  IOException {
+    final FileSystem fs = FileSystem.get(yarnConfiguration);
+    final Path httpEndpointPath = new Path(dfsPath, fileNames.getDriverHttpEndpoint());
+
+    String trackingUri = null;
+    for (int i = 0; i < 60; i++) {
+      try {
+        LOG.log(Level.INFO, "Attempt " + i + " reading " + httpEndpointPath.toString());
+        if (fs.exists(httpEndpointPath)) {
+          FSDataInputStream input = fs.open(httpEndpointPath);
+          BufferedReader reader = new BufferedReader(new InputStreamReader(input));
+          trackingUri = reader.readLine();
+          reader.close();
+          break;
+        }
+      } catch (Exception ex) {
+      }
+      try{
+        Thread.sleep(1000);
+      } catch(InterruptedException ex2) {
+        break;
+      }
+    }
+
+    if (null == trackingUri) {
+      trackingUri = "";
+      LOG.log(Level.WARNING, "Failed reading " + httpEndpointPath.toString());
+    }
+
+    final File driverHttpEndpointFile = new File(driverFolder, fileNames.getDriverHttpEndpoint());
+    BufferedWriter out = new BufferedWriter(new FileWriter(driverHttpEndpointFile));
```

Comment:
Same Findbugs warning here; `new OutputStreamWriter(file, "UTF-8")` instead of `FileWriter` should fix it